src/data/sqlite_converter.py

```diff
@@ -0,0 +1,389 @@
+    files_list = []
+    GCD_list   = []
+    root,folders,root_files = next(os.walk(dir))
+    gcds_root = []
+    gcd_root = None
+    i3files_root = []
+    for file in root_files:
+        if has_extension(file, extensions):
+            if isi3(file):
+                i3files_root.append(os.path.join(root,file))
+            else:
+                gcd_root = os.path.join(root,file)
+                gcds_root.append(os.path.join(root,file))
+    if gcd_root == None:
+        gcd_root = gcd_rescue
+    for k in range(len(i3files_root) - len(gcds_root)):
+        gcds_root.append(gcd_root)
+    files_list.extend(i3files_root)
+    GCD_list.extend(gcds_root)
+    for folder in folders:
+        sub_root, sub_folders, sub_folder_files = next(os.walk(os.path.join(root,folder)))
+        gcds_folder = []
+        gcd_folder = None
+        i3files_folder = []
+        for sub_folder_file in sub_folder_files:
+            if has_extension(sub_folder_file, extensions):
+                if isi3(sub_folder_file):
+                    i3files_folder.append(os.path.join(sub_root,sub_folder_file))
+                else:
+                    gcd_folder = os.path.join(sub_root,sub_folder_file)
+                    gcds_folder.append(os.path.join(sub_root,sub_folder_file))
+        if gcd_folder == None:
+            gcd_folder = gcd_rescue
+        for k in range(len(i3files_folder) - len(gcds_folder)):
+            gcds_folder.append(gcd_folder)
+        files_list.extend(i3files_folder)
+        GCD_list.extend(gcds_folder)
+    
+    files_list, GCD_list = pairwiseshuffle(files_list, GCD_list)
+    return files_list, GCD_list
+
+def pairwiseshuffle(files_list, gcd_list):
+    df = pd.DataFrame({'i3': files_list, 'gcd': gcd_list})
+    df_shuffled = df.sample(frac = 1)
+    i3_shuffled = df_shuffled['i3'].tolist()
+    gcd_shuffled = df_shuffled['gcd'].tolist()
+    return i3_shuffled, gcd_shuffled
+
+
+def find_files(paths,outdir,db_name,gcd_rescue, extensions = None):
+    print('Counting files in: \n%s\n This might take a few minutes...'%paths)
+    if extensions == None:
+        extensions = ("i3.bz2",".zst",".gz")
+    input_files_mid = []
+    input_files = []
+    files = []
+    gcd_files_mid = []
+    gcd_files = []
+    for path in paths:
+        input_files_mid, gcd_files_mid = walk_directory(path, extensions, gcd_rescue)
+        input_files.extend(input_files_mid)
+        gcd_files.extend(gcd_files_mid)
+
+    if len(input_files) > 0:
+        input_files, gcd_files = pairwiseshuffle(input_files, gcd_files)
+        save_filenames(input_files, outdir, db_name)
+    return input_files, gcd_files
+    
+def save_filenames(input_files,outdir, db_name):
+    create_out_directory(outdir + '/%s/config'%db_name)
+    input_files = pd.DataFrame(input_files)
+    input_files.columns = ['filename']
+    input_files.to_csv(outdir + '/%s/config/i3files.csv'%db_name)
+    return
+
+def create_out_directory(outdir):
```

Comment:
The return value of this function does not seem to be used anywhere. Given that, is this function strictly necessary or could it be replaced by calls to `os.makedirs(outdir, exists_ok=True)`? Otherwise, perhaps drop the return values from this function and consider producing a slightly less dire warning. ;)

Reply:
Return value is used at least once. But the check is not strictly neccesary. 

The method you mention is not compatible with python >3.0 and by using error handling we get the same functionality.